src/components/RJST/components/Table/TableActions.tsx

```diff
@@ -68,35 +108,72 @@ export const TableActions = <T extends object>({
 				anchorEl={anchorEl}
 				open={open}
 				onClose={handleClose}
+				slotProps={{
+					paper: {
+						sx: {
+							minWidth: 250,
+						},
+					},
+				}}
 			>
-				<FormGroup>
-					{columns.map((column) => (
-						<MenuItem key={column.key} sx={{ p: 0 }}>
-							{/* Needed to expand the checkbox click to MenuItem */}
-							<FormControlLabel
-								sx={{
-									flex: 1,
-									width: '100%',
-									height: '100%',
-									py: 1,
-									px: 2,
-									m: 0,
-								}}
-								control={
-									<Checkbox
-										onClick={() => {
-											handleColumnSelection(column);
-										}}
-										checked={column.selected}
-									/>
-								}
-								label={
-									typeof column.label === 'string' ? column.label : column.title
-								}
-							/>
-						</MenuItem>
-					))}
-				</FormGroup>
+				<DragDropContext onDragEnd={handleDragEnd}>
+					<Droppable droppableId="columns" direction="vertical">
+						{(provided) => (
+							<FormGroup ref={provided.innerRef} {...provided.droppableProps}>
+								{columns.map((column, index) => (
+									<Draggable
+										key={column.key}
+										draggableId={column.key}
+										index={index}
+									>
+										{(item) => (
+											<MenuItem
+												ref={item.innerRef}
+												{...item.draggableProps}
+												sx={{ display: 'flex', alignItems: 'center', p: 0 }}
+											>
+												{analyticsState.featureFlags?.columnOrdering && (
+													<ListItemIcon
```

Comment:
Why `ListItemIcon` instead of the icon alone? Necessary for `item.dragHandleProps`?

Reply:
mainly accessibility, in a list (MenuItem in this case) that's what is suggested to respect MUI's semantic structure

eg: https://mui.com/material-ui/react-menu/